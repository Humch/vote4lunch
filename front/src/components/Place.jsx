import React, { Component } from 'react';
import { connect } from 'react-redux';

import { ListGroupItem, Button } from 'reactstrap'

import { deleteAPlace } from '../actions/deleteAPlace'

import MaterialIcon from 'material-icons-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTimes } from '@fortawesome/free-solid-svg-icons'


class Place extends Component {
    constructor(props) {
        super(props);
        this.deleteAPlaceToVote = this.deleteAPlaceToVote.bind(this);
    }

    deleteAPlaceToVote(vote_id, place_id) {
        this.props.dispatch(deleteAPlace(vote_id, place_id))
    }

    render() {

        const { place, type, vote_id, place_id } = this.props

        return (
            <ListGroupItem>
                <p className="my-1">
                    {type === 'restaurant' ? <MaterialIcon icon="restaurant" /> : <MaterialIcon icon="fastfood" />} {place}
                    <span className="align-top">
                        <Button
                            color="danger"
                            size="sm"
                            className="float-right"
                            onClick={() => this.deleteAPlaceToVote(vote_id, place_id) }
                        >
                            <FontAwesomeIcon icon={faTimes} />
                        </Button>
                    </span>

                </p>
            </ListGroupItem>
        );
    }
}

const mstp = () => ({
});

export default connect(mstp)(Place);
