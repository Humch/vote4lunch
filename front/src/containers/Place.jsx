import React, { Component } from 'react';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { ListGroupItem, Button } from 'reactstrap'

import { deleteAPlace } from '../actions/deleteAPlace'

import MaterialIcon from 'material-icons-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTimes } from '@fortawesome/free-solid-svg-icons'


class Place extends Component {
    constructor(props) {
        super(props);
    }

    render() {

        const { place, type, vote_id, place_id, deleteAPlace } = this.props

        return (
            <ListGroupItem>
                <p className="my-1">
                    {type === 'restaurant' ? <MaterialIcon icon="restaurant" /> : <MaterialIcon icon="fastfood" />} {place}
                    <span className="align-top">
                        <Button
                            color="danger"
                            size="sm"
                            className="float-right"
                            onClick={() => deleteAPlace(vote_id, place_id) }
                        >
                            <FontAwesomeIcon icon={faTimes} />
                        </Button>
                    </span>

                </p>
            </ListGroupItem>
        );
    }
}

const mdtp = (dispatch) => {
    return bindActionCreators({ deleteAPlace }, dispatch);
}

export default connect(null, mdtp)(Place);
