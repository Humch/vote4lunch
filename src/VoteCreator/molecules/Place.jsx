import MaterialIcon from 'material-icons-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

import { ListGroupItem, Button } from 'reactstrap';

import { deleteAPlace } from '../../actions/deleteAPlace';

const Place = ({ place, type, vote_id, place_id, deleteAPlace: dAP }) => (
  <ListGroupItem>
    <p className="my-1">
      {type === 'restaurant' ? <MaterialIcon icon="restaurant" /> : <MaterialIcon icon="fastfood" />} {place}
      <span className="align-top">
        <Button
          color="danger"
          size="sm"
          className="float-right"
          onClick={() => dAP(vote_id, place_id)}
        >
          <FontAwesomeIcon icon={faTimes} />
        </Button>
      </span>

    </p>
  </ListGroupItem>
);

Place.propTypes = {
  place: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  vote_id: PropTypes.string.isRequired,
  place_id: PropTypes.string.isRequired,
  deleteAPlace: PropTypes.func.isRequired,
};

const mdtp = dispatch => bindActionCreators({ deleteAPlace }, dispatch);

export default connect(null, mdtp)(Place);
