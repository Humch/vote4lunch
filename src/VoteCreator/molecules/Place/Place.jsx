import MaterialIcon from 'material-icons-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

import { ListGroupItem, Button } from 'reactstrap';

import { deleteAPlace } from '../../actions';

const Place = ({
  placeName,
  type,
  voteId,
  placeId,
  deleteAPlace: dAP,
}) => (
  <ListGroupItem>
    <p className="my-1">
      {type === 'restaurant' ? <MaterialIcon icon="restaurant" /> : <MaterialIcon icon="fastfood" />}
      { '' }
      {placeName}
      <span className="align-top">
        <Button
          color="danger"
          size="sm"
          className="float-right"
          onClick={() => dAP(voteId, placeId)}
        >
          <FontAwesomeIcon icon={faTimes} />
        </Button>
      </span>

    </p>
  </ListGroupItem>
);

Place.propTypes = {
  placeName: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  voteId: PropTypes.number.isRequired,
  placeId: PropTypes.number.isRequired,
  deleteAPlace: PropTypes.func.isRequired,
};

const mdtp = dispatch => bindActionCreators({ deleteAPlace }, dispatch);

export default connect(null, mdtp)(Place);
