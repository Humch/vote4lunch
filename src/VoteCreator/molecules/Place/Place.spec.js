/* eslint-disable no-undef */
import React from 'react';
import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Place from './Place';

describe('Place Snapshot', () => {
  const initialState = {};
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of Place with restaurant icon', () => {
    const renderedValue = renderer.create(
      <Place
        store={store}
        placeName="Cafétéria Universitaire Le Veracruz"
        type="restaurant"
        voteId={1}
        placeId={1}
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('capturing Snapshot of Place with fastfood icon', () => {
    const renderedValue = renderer.create(
      <Place
        store={store}
        placeName="Cafétéria Universitaire Le Veracruz"
        type="fastfood"
        voteId={1}
        placeId={1}
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('Place - REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = {};
  const middlewares = [reduxThunk];
  const mockStore = configureStore(middlewares);
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);

    wrapper = mount(
      <Provider store={store}>
        <Place
          placeName="Cafétéria Universitaire Le Veracruz"
          type="restaurant"
          voteId={1}
          placeId={1}
        />
      </Provider>
    );
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check if Button delete the place', () => {
    const buttonDelete = wrapper.find('Button');
    buttonDelete.simulate('click');
  });
});
