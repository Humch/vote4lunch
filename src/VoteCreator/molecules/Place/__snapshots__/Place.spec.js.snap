// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place Snapshot capturing Snapshot of Place with fastfood icon 1`] = `
<li
  className="list-group-item"
>
  <p
    className="my-1"
  >
    <i
      className="material-icons md-24 md-dark "
      icon="fastfood"
      style={
        Object {
          "color": "",
          "fontSize": "md-24",
        }
      }
    />
    
    Cafétéria Universitaire Le Veracruz
    <span
      className="align-top"
    >
      <button
        aria-label={null}
        className="float-right btn btn-danger btn-sm"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </span>
  </p>
</li>
`;

exports[`Place Snapshot capturing Snapshot of Place with restaurant icon 1`] = `
<li
  className="list-group-item"
>
  <p
    className="my-1"
  >
    <i
      className="material-icons md-24 md-dark "
      icon="restaurant"
      style={
        Object {
          "color": "",
          "fontSize": "md-24",
        }
      }
    />
    
    Cafétéria Universitaire Le Veracruz
    <span
      className="align-top"
    >
      <button
        aria-label={null}
        className="float-right btn btn-danger btn-sm"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </span>
  </p>
</li>
`;
